src/sketch.js

```diff
@@ -1,45 +1,81 @@
 export default function sketch(p) {
-    let loopAllowed = false;
-    let mapWidth = 30;
-    let mapHeight = 30;
-    let zoomLevel = 1;
-
-    let dragMode = "move";
-
-    let mapArray = [];
-
+    let loopAllowed = false,
+        mapWidth = 30,
+        mapHeight = 30,
+        zoomLevel = 1,
+        dragMode = "move",
+        mapArray = [],
+        pixelOffsetX = 0,
+        pixelOffsetY = 0,
+        movingCanvas = false,
+        dragging = false,
+        
+        ghostFigure = null,
 
-    const tileSize = 10;
-    let pixelOffsetX = 0,
-        pixelOffsetY = 0;
+        activeAsset = null,
+        activeImage = null,
+        activeImageLabel = null
+    ;
+    const tileSize = 100;
+        // cursorTileSize = 50;
 
     p.setup = function () {
         const parentEl = document.getElementById('mapEditor');
-        console.log({parentEl, clW: parentEl.clientWidth, clH: parentEl.clientHeight});
         p.createCanvas(parentEl.clientWidth, parentEl.clientHeight);
 
-        pixelOffsetX = p.width / 2
+        // TODO: Replace this with code that centers the field
+        pixelOffsetX = 100;
+        pixelOffsetY = 100;
+        zoomLevel = 0.5;
 
         loopAllowed = true;
 
-        for (let i = 0; i < mapHeight; i++)
+        for (let i = 0; i < mapHeight; i++) {
             mapArray.push(new Array(mapWidth));
-
-        console.table(mapArray);
+        }
     };
 
+    const imgCache = {};
+    function loadImage(name) {
+        return new Promise((res, rej) => {
+            if (imgCache[name]) return res(imgCache[name]);
+
+            p.loadImage(`/images/objects/${name}.jpg`, img => {
+                imgCache[name] = img;
+                res(img);
+            });
+        });
+    }
+
     p.myCustomRedrawAccordingToNewPropsHandler = function (props) {
         console.log('myCustomRedrawAccordingToNewPropsHandler', props);
+
+        if (props.activeAsset) {
+            loadImage(props.activeAsset.img)
+                .then(img => {activeImage = img; activeImageLabel = props.activeAsset.img})
+                .catch(console.error);
+        }
     };
 
     p.draw = function () {
         if (!loopAllowed) {
             return;
         }
 
-        // analyzeCursor();
+        analyzeMouse();
+        analyzeKeyboard();
         p.background(150);
 
+        if (ghostFigure) {
+            loadImage(ghostFigure.img)
+                .then(img => {
+                    p.push();
+                        p.tint(255, 128);
```

Comment:
Сейчас если в клетке уже нарисован блок, то похожий по цвету «призрак» сливается с фоном. Зелёная подсветка эту проблему решает.
```suggestion
                        p.tint(0, 256, 0, 128);
```